holmes/core/conversations.py

```diff
@@ -200,6 +200,8 @@ def build_issue_chat_messages(
     investigation_analysis = issue_chat_request.investigation_result.result
     tools_for_investigation = issue_chat_request.investigation_result.tools
 
+    enabled_toolsets = [ts.name for ts in ai.tool_executor.enabled_toolsets]
```

Comment:
consider to move it as property/function to the tool_executor instead of recreate this list everytime.

Also instead of list please change to set since we are using in operator on it, that way it won't need to scan the list.